rust/kernel/hw_random.rs

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Hardware Random Number Generator
+//!
+//! C header: [`include/linux/hw_random.h`](../../../../include/linux/hw_random.h)
+
+use alloc::{boxed::Box, slice::from_raw_parts_mut};
+
+use crate::{
+    bindings, c_types, from_kernel_result, str::CStr, types::PointerWrapper, Error, Result,
+};
+
+use core::{cell::UnsafeCell, marker::PhantomData, ops::Deref, pin::Pin};
+
+/// This trait is implemented in order to provide callbacks to `struct hwrng`.
+pub trait HwrngOperations: Sized {
+    /// The methods to use to populate [`struct hwrng`].
+    const TO_USE: ToUse;
+
+    /// The pointer type that will be used to hold user-defined data type.
+    type Data: PointerWrapper = Box<Self>;
+
+    /// Initialization callback (can be NULL).
+    fn init(_data: &<<Self::Data as PointerWrapper>::Borrowed as Deref>::Target) -> Result {
+        Err(Error::EINVAL)
+    }
+
+    /// Cleanup callback (can be NULL).
+    fn cleanup(_data: &<<Self::Data as PointerWrapper>::Borrowed as Deref>::Target) {}
+
+    /// New API. drivers can fill up to max bytes of data
+    /// into the buffer. The buffer is aligned for any type
+    /// and max is a multiple of 4 and >= 32 bytes.
+    fn read(
+        _data: &<<Self::Data as PointerWrapper>::Borrowed as Deref>::Target,
+        _buffer: &mut [i8],
+        _wait: bool,
```

Comment:
No need to prefix with `_` if it is just a declaration:

```suggestion
        wait: bool,
```

Reply:
ack, thanks